src/neuralnet/superblock.h

```diff
@@ -0,0 +1,248 @@
+#pragma once
+
+#include "scraper/fwd.h"
+#include <string>
+#include "key.h"
+#include "neuralnet/cpid.h"
+
+std::string UnpackBinarySuperblock(std::string block);
+std::string PackBinarySuperblock(std::string sBlock);
+
+namespace NN {
+
+// TODO: move this out into a common location...?
+template<typename integer_t>
+struct Compact
+{
+    Compact() : m_value(0)
+    {
+    }
+
+    Compact(integer_t value) : m_value(value)
+    {
+    }
+
+    integer_t m_value;
+
+    // TODO: implement the other general-purpose operators...
+
+    integer_t operator*() const
+    {
+        return m_value;
+    }
+
+    bool operator<(Compact<integer_t> other) const
+    {
+        return m_value < other.m_value;
+    }
+
+    bool operator<(integer_t other) const
+    {
+        return m_value < other;
+    }
+
+    void operator=(integer_t value)
+    {
+        m_value = value;
```

Comment:
I think we want to return a reference to `m_value` here, see http://www.icu-project.org/docs/papers/cpp_report/the_anatomy_of_the_assignment_operator.html:

> An assignment operator can return anything it wants, but the standard C and C++ assignment operators return a reference to the left-hand operand.

Reply:
Noted. Will update if we keep `Compact<T>` in favor of `VarInt`.

Reply:
This is not an issue anymore, right?

Reply:
That's right--the latest version removed this class.